src/io/fasta.rs

```diff
@@ -96,29 +97,30 @@ impl<R: io::Read> Reader<R> {
 
 /// A FASTA index as created by SAMtools (.fai).
 pub struct Index {
-    inner: collections::HashMap<String, IndexRecord>,
-    seqs: Vec<String>,
+    inner: Vec<IndexRecord>,
+    name_to_rid: collections::HashMap<String, usize>,
```

Comment:
This looks perfect. Now, if you add the name additionally to the IndexRecord, the sequences method becomes very easy, without sorting.